Updating mbed-coap to version 4.0.8

yogpan01 opened the pull request:
## Description
Fixes errors reported in Github 
 - https://github.com/ARMmbed/mbed-os/issues/4798
    - Fixed Lwm2m blockwise data transfer (using Block1 option) with Leshan
 - https://github.com/ARMmbed/mbed-client/issues/512
    -  Fixed an ongoing blocktransfer (Block 1 option) interruption by out of order Block to return error 4.08 Request Entity Incomplete as specified in the coap specification. 
 - https://github.com/ARMmbed/mbed-client/issues/511
   - Fixed detected duplicate message duplications will not stop mbed-client

## Status
**READY**


## Migrations
If this PR changes any APIs or behaviors, give a short description of what *API users* should do when this PR is merged.

NO


## Related PRs
List related PRs against other branches:
https://github.com/ARMmbed/mbed-coap/pull/29
https://github.com/ARMmbed/mbed-coap/pull/28
https://github.com/ARMmbed/mbed-coap/pull/25

theotherjimmy commented:
@yogpan01 We use the PR titles is release notes, and I would not expect anyone to know what anything IOTCLT-* means. Could you update the title to reflect what you're fixing?

0xc0170 commented:
> @yogpan01 We use the PR titles is release notes, and I would not expect anyone to know what anything IOTCLT-* means. Could you update the title to reflect what you're fixing?

Same for the commit message 👍

yogpan01 commented:
@0xc0170 I have updated the PR title, the PR is fix for the issues reported in Github and I have updated the issue number in the title as well as in comment section.

theotherjimmy commented:
@yogpan01 We recommend that you describe what you are fixing instead of referencing issues, as this is easier to understand it the release notes. Could you add a short description of what you are fixing in the title and commit messages instead of the issue numbers? It looks like you updated the `mbed-coap` library, so something like `Update mbed-coap to version 4.0.8` might be easier to understand.

0xc0170 commented:
> @yogpan01 We recommend that you describe what you are fixing instead of referencing issues, as this is easier to understand it the release notes. Could you add a short description of what you are fixing in the title and commit messages instead of the issue numbers? It looks like you updated the mbed-coap library, so something like Update mbed-coap to version 4.0.8 might be easier to understand.

Can you please update? We will then start the CI

yogpan01 commented:
@anttiylitokola Can you please update the PR title and description as requested by @0xc0170 ?

anttiylitokola commented:
@yogpan01, This is data is coming from your own fork so I do not have rights to update anything here.

yogpan01 commented:
@0xc0170 I have updated the title and also added some information about the actual issues that were fixed in description. Is this ok now ?

0xc0170 commented:
> @0xc0170 I have updated the title and also added some information about the actual issues that were fixed in description. Is this ok now ?

Almost, the most important part is being it part of the commit (github description then can be the same). if you can ammend the commit to include that info, would be appreciated. helps a lot

yogpan01 commented:
@0xc0170 Done the update on the commit. Please let me know, if this is fine now.

0xc0170 commented:
/morph test
